### ISSUE TITLE
improve const correctness like in the example `Mat_VarCreate`

### ISSUE THREAD

**Issue by Febbe** [tbeu/matio#137]
change 
```
EXTERN matvar_t  *Mat_VarCreate(const char *name,enum matio_classes class_type,
                      enum matio_types data_type,int rank,size_t *dims,
                      void *data,int opt);
```
To
```
EXTERN matvar_t  *Mat_VarCreate(const char *name, enum matio_classes class_type,
                      enum matio_types data_type, int rank, size_t const*dims,
                      void const*data,int opt);
```

There are also other places where this is necessary.

**Comment by tbeu** [comment]
This touches the API which I won't do at the moment. You still can update it in matio.h and use in your application w/o further issue, I believe.

**Comment by Febbe** [comment]
Right, I see your point. The only thing which would work, is adding those functions and mark the other as deprecated. For now I'll just const_cast<void>(data) them. But in some unlucky cases, this could lead to undefined behavior.

**Comment by tbeu** [comment]
Finally resolved by dde072ee0bb4a5040d45bcb90c1ea6c0b468f283.
